replica/table.cc

```diff
@@ -1073,10 +1073,14 @@ table::stop() {
     if (_async_gate.is_closed()) {
         co_return;
     }
-    co_await _async_gate.close();
+    // Allow `compaction_manager::stop` to stop ongoing compactions
+    // while they may still hold the _async_gate
+    // entered in `compaction_manager::start_compaction`
```

Comment:
`compaction_manager::start_compaction` enters `compaction_state::gate`, but here `table::_async_gate` is closed

Reply:
right, I'll revise the comment

Reply:
simplified the comment in v2